src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -1373,6 +1378,89 @@ switches"() {
         !isSwitchValid && switches.each { northbound.synchronizeSwitch(it, true)}
     }
 
+    @Tidy
+    def "Able to swap endpoints for a flow with flowLoop"() {
+        setup: "Create two flows with the same src and different dst switches"
+        def tgSwitchIds = topology.getActiveTraffGens()*.switchConnected*.dpId
+        assumeTrue("Not enough traffgen switches found", tgSwitchIds.size() > 1)
+        SwitchPair flow2SwitchPair = null
+        SwitchPair flow1SwitchPair = topologyHelper.getAllNeighboringSwitchPairs().find { firstPair ->
+            def firstOk = firstPair.src.dpId in tgSwitchIds && firstPair.dst.dpId in tgSwitchIds
+            flow2SwitchPair = topologyHelper.getAllNeighboringSwitchPairs().find { secondPair ->
+                secondPair.src.dpId == firstPair.src.dpId && secondPair.dst.dpId != firstPair.dst.dpId
+            }
+            firstOk && flow2SwitchPair
+        }
+        assumeTrue("Required switch pairs not found in given topology",
+                flow1SwitchPair.asBoolean() && flow2SwitchPair.asBoolean())
+
+        def flow1 = flowHelper.randomFlow(flow1SwitchPair, true)
+        def flow2 = flowHelper.randomFlow(flow2SwitchPair, true, [flow1])
+
+        flowHelper.addFlow(flow1)
+        flowHelper.addFlow(flow2)
+
+        and: "FlowLoop is created for the second flow on the dst switch"
+        northboundV2.createFlowLoop(flow2.id, new FlowLoopPayload(flow2SwitchPair.dst.dpId))
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getFlowStatus(flow2.id).status == FlowState.UP
+        }
+
+        when: "Try to swap endpoints for two flows"
+        def flow1Dst = flow2.destination
+        def flow1Src = flow2.source
+        def flow2Dst = flow1.destination
+        def flow2Src = flow1.source
```

Comment:
such swap is not typical. It doesn't make sense to swap both src and dst so that flows just swap their flowIds. Usually flows swap either their src and retain original dst, or vice versa